playground/backend/internal/db/datastore/datastore_db.go

```diff
@@ -236,11 +237,44 @@ func (d *Datastore) GetCatalog(ctx context.Context, sdkCatalog []*entity.SDKEnti
 		return nil, err
 	}
 
+	//Retrieving datasets
+	datastoreQuery := datastore.NewQuery(constants.DatasetsKind).Namespace(utils.GetNamespace(ctx))
```

Comment:
this query gets ALL the datasets, which isn't scalable: it will soon become a bottleneck